### PR TITLE
[Feature][Feature/AES-1786 ] Notification Modal

### PR DESCRIPTION
**Pull request by zack76**
https://aesoponline.atlassian.net/browse/AES-1876

New components: Notification Modal & Context for Notification for further use in NewsLetterSignUp component

The notification modal will cover the whole footer while will be triggered by only the email subscription button which is the child component of the footer.

ezgif-6-3a682a818a0c

### CODE REVIEW

File: src/components/NotificationModal/NotificationModal.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import Icon from '~/components/Icon';
+import styles from './NotificationModal.module.css';
+import { useNotificationContext } from 'contexts';
```
Comment (beau-charman) [inline src/components/NotificationModal/NotificationModal.js:6]:
```
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { useNotificationContext } from 'contexts';
import Icon from '~/components/Icon';
import styles from './NotificationModal.module.css';
```

File: src/components/NotificationModal/NotificationModal.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+const NotificationModal = ({
+  className,
+  backgroundColor,
+  notificationMessage,
```
Comment (beau-charman) [inline src/components/NotificationModal/NotificationModal.js:11]:
```  
  backgroundColor,  
  className,
  notificationMessage,
```

File: src/components/NotificationModal/NotificationModal.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+}) => {
+  const footerSuccessModal = useNotificationContext();
+  const { actionType, dispatch, showModal } = footerSuccessModal;
+
+  const onClick = e => {
```
Comment (beau-charman) [inline src/components/NotificationModal/NotificationModal.js:16]: when defining a event handler, prefix with `handle`, so `handleOnClick `

File: src/components/NotificationModal/NotificationModal.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        <Icon height={15} name="close" theme="dark" width={15} />
+      </button>
+    </div>
+  );
+};
+
+NotificationModal.propTypes = {
+  className: PropTypes.string,
+  backgroundColor: PropTypes.string,
+  notificationMessage: PropTypes.string,
```
Comment (beau-charman) [inline src/components/NotificationModal/NotificationModal.js:47]:
```
  backgroundColor: PropTypes.string,
  className: PropTypes.string,
  notificationMessage: PropTypes.string,
```

File: src/components/NotificationModal/NotificationModal.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+NotificationModal.defaultProps = {
+  className: undefined,
+  backgroundColor: undefined,
+  notificationMessage: undefined,
```
Comment (beau-charman) [inline src/components/NotificationModal/NotificationModal.js:53]:
```
  backgroundColor: undefined,
  className: undefined,
  notificationMessage: undefined,
```

File: src/components/NotificationModal/ToogleButton.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
```
Comment (beau-charman) [inline src/components/NotificationModal/ToogleButton.js:1]: any child components for a parent component, please place them within a subsequent `components` folder. See `Carousel`

File: src/styles/variables/layout.css [hunk 7]
```diff
@@ -8,6 +8,9 @@
   --layout-logo-vertical-offset-md: 175px;
   --layout-logo-vertical-offset-lg: 200px;
   --layout-menu-height: 80px;
+  --site-lg-margin: 40px;
+  --site-md-margin: 34px;
+  --site-sm-margin: 20px;
```
Comment (beau-charman) [inline src/styles/variables/layout.css:13]: could we keep the `layout` prefix here? Makes it easy to know where the variable is kept. 

```
  --layout-lg-margin: 40px;
  --layout-md-margin: 34px;
  --layout-sm-margin: 20px;
```

**Review by beau-charman — CHANGES_REQUESTED** [review]

**Comment by beau-charman** [comment]
Nice stuff @zack76. Just a couple of things to address.

File: src/components/NotificationModal/NotificationModal.js (hunk 1 above)
Comment (zack76) [reply src/components/NotificationModal/NotificationModal.js]: updated

File: src/components/NotificationModal/NotificationModal.js (hunk 2 above)
Comment (zack76) [reply src/components/NotificationModal/NotificationModal.js:11]: updated

File: src/components/NotificationModal/NotificationModal.js (hunk 3 above)
Comment (zack76) [reply src/components/NotificationModal/NotificationModal.js]: updated

File: src/components/NotificationModal/NotificationModal.js (hunk 4 above)
Comment (zack76) [reply src/components/NotificationModal/NotificationModal.js:47]: updated

File: src/components/NotificationModal/NotificationModal.js (hunk 5 above)
Comment (zack76) [reply src/components/NotificationModal/NotificationModal.js:53]: updated

File: src/styles/variables/layout.css (hunk 7 above)
Comment (zack76) [reply src/styles/variables/layout.css]: updated

File: src/components/NotificationModal/ToogleButton.js (hunk 6 above)
Comment (zack76) [reply src/components/NotificationModal/ToogleButton.js]: updated

**Review by beau-charman — APPROVED** [review]
you boss Zack!

**Pull request merged**